src/query/api/v1/handler/database/common.go

```diff
@@ -43,31 +40,23 @@ type Handler struct {
 
 // RegisterRoutes registers the namespace routes
 func RegisterRoutes(
-	r *mux.Router,
+	addRoute func(path string, handler http.Handler, methods ...string),
```

Comment:
nit: extract as a type, e.g. `type AddRouteFn func(path string, handler http.Handler, methods ...string)`

Reply:
Actually, the `...string` seems like it may be a little fragile, what would you think of making methods a `[]string`? Will be more annoying to use but a little clearer what we're doing, probably up to you